build.gradle

```diff
@@ -13,48 +13,38 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import org.gradle.api.internal.plugins.osgi.OsgiHelper
-
 buildscript {
+  //we define kotlin version for benefit of both core and test (see kotlin-gradle-plugin below)
   ext.kotlinVersion = '1.3.31'
   repositories {
 	maven { url "https://repo.spring.io/plugins-release" }
   }
   dependencies {
-	classpath 'org.springframework.build.gradle:propdeps-plugin:0.0.7'
-	classpath 'io.spring.gradle:spring-io-plugin:0.0.4.RELEASE'
-	classpath "com.github.jengelman.gradle.plugins:shadow:4.0.2"
+	classpath 'org.springframework.build.gradle:propdeps-plugin:0.0.7' //still uses the old ways
 	classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${kotlinVersion}"
   }
 }
 
 plugins {
+  id "com.github.johnrengelman.shadow" version "4.0.2"
   id 'org.asciidoctor.convert' version '1.5.11'
   id "me.champeau.gradle.japicmp" version "0.2.6"
   id "de.undercouch.download" version "3.4.3"
   id "org.unbroken-dome.test-sets" version "2.1.1" apply false // 2.x version does not work with Gradle 4.10
+  id 'com.gradle.build-scan' version '2.4.2'
 }
 
-apply from: "gradle/doc.gradle"
+apply plugin: "io.reactor.gradle.detect-ci"
+apply from: "gradle/asciidoc.gradle" //asciidoc (which is generated from root dir)
 apply from: "gradle/setup.gradle"
 
-
 ext {
-  //also set up the special version at root, for refguide
-  if (project.hasProperty('versionBranch') && version.toString().endsWith(".BUILD-SNAPSHOT")) {
-	versionBranch = versionBranch.replaceAll("\"", "").trim()
-	if (!versionBranch.isEmpty()) {
-	  realVersion = version.toString().replace("BUILD-SNAPSHOT", versionBranch + ".BUILD-SNAPSHOT")
-	  project.version = realVersion
-	}
-  }
-
-  def ciServers = ["CI", "CONTINUOUS_INTEGRATION", "TRAVIS", "CIRCLECI", "bamboo_planKey"]
-  ciServers.retainAll(System.getenv().keySet())
-  isCiServer = !ciServers.isEmpty()
-  if (isCiServer) {
-	println "Detected CI environment: $ciServers"
+  jdk = JavaVersion.current().majorVersion
+  jdkJavadoc = "https://docs.oracle.com/javase/$jdk/docs/api/"
+  if (JavaVersion.current().isJava11Compatible()) {
+	jdkJavadoc = "https://docs.oracle.com/en/java/javase/$jdk/docs/api/"
```

Comment:
the indentation seems to be wrong 
<img width="592" alt="Screenshot 2019-10-22 at 5 23 16 PM" src="https://user-images.githubusercontent.com/1050762/67302094-a7706480-f4f0-11e9-93f4-b0e4c9b5b2d6.png">

Reply:
I'll fix modified lines to use tabs consistently